src/Microsoft.Health.Dicom.Core/Models/DicomIdentifier.cs

```diff
@@ -0,0 +1,252 @@
+        => new DicomIdentifier(
+            studyInstanceUid,
+            EnsureArg.IsNotNullOrWhiteSpace(seriesInstanceUid, nameof(seriesInstanceUid)),
+            EnsureArg.IsNotNullOrWhiteSpace(sopInstanceUid, nameof(sopInstanceUid)));
+
+    /// <summary>
+    /// Creates a new <see cref="DicomIdentifier"/> structure equivalent to the given <see cref="VersionedInstanceIdentifier"/>.
+    /// </summary>
+    /// <param name="identifier">A versioned instance identifier.</param>
+    /// <returns>A <see cref="DicomIdentifier"/> structure for the given SOP instance.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="identifier"/> is <see langword="null"/>.</exception>
+    public static DicomIdentifier ForInstance(VersionedInstanceIdentifier identifier)
+        => new DicomIdentifier(
+            EnsureArg.IsNotNull(identifier, nameof(identifier)).StudyInstanceUid,
+            identifier.SeriesInstanceUid,
+            identifier.SopInstanceUid);
+
+    /// <summary>
+    /// Converts the string representation of a <see cref="DicomIdentifier"/> to its equivalent structure representation.
+    /// </summary>
+    /// <param name="value">A string that contains a <see cref="DicomIdentifier"/> to convert.</param>
+    /// <returns>
+    /// An object that is equivalent to the <see cref="DicomIdentifier"/> contained in <paramref name="value"/>.
+    /// </returns>
+    /// <exception cref="ArgumentNullException"><paramref name="value"/> is <see langword="null"/>.</exception>
+    /// <exception cref="FormatException">
+    /// <paramref name="value"/> does not contain a valid string representation of a <see cref="DicomIdentifier"/>.
+    /// </exception>
+    public static DicomIdentifier Parse(string value)
+    {
+        string[] parts = EnsureArg.IsNotNull(value, nameof(value)).Split('/', StringSplitOptions.TrimEntries);
+        if (parts.Length == 0 || parts.Length > 3)
+            throw new FormatException(string.Format(CultureInfo.CurrentCulture, DicomCoreResource.InvalidDicomIdentifier, value));
+
+        string studyInstanceUid, seriesInstanceUid = null, sopInstanceUid = null;
+
+        // Parse and Validate
+        try
+        {
+            studyInstanceUid = parts[0];
+            UidValidation.Validate(studyInstanceUid, nameof(StudyInstanceUid));
+
+            if (parts.Length > 1)
+            {
+                seriesInstanceUid = parts[1];
+                UidValidation.Validate(seriesInstanceUid, nameof(SeriesInstanceUid));
+
+                if (parts.Length == 3)
+                {
+                    sopInstanceUid = parts[2];
+                    UidValidation.Validate(sopInstanceUid, nameof(SopInstanceUid));
+                }
+            }
+        }
+        catch (InvalidIdentifierException iie)
+        {
+            throw new FormatException(string.Format(CultureInfo.CurrentCulture, DicomCoreResource.InvalidDicomIdentifier, value), iie);
```

Comment:
Is this caught has know exception in the Exceptionmiddleware? OR does this need to be wrapped on a validationException?

Reply:
Oh, good catch. I'll check this